db-async-common/src/main/java/com/github/jasync/sql/db/column/LocalDateTimeEncoderDecoder.kt

```diff
@@ -1,27 +1,25 @@
 package com.github.jasync.sql.db.column
 
-import org.joda.time.LocalDateTime
-import org.joda.time.format.DateTimeFormatterBuilder
+import com.github.jasync.sql.db.util.microsecondsFormatter
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatterBuilder
 
 object LocalDateTimeEncoderDecoder : ColumnEncoderDecoder {
 
     private const val ZeroedTimestamp = "0000-00-00 00:00:00"
 
-    private val optional = DateTimeFormatterBuilder()
-        .appendPattern(".SSSSSS").toParser()
-
     private val format = DateTimeFormatterBuilder()
         .appendPattern("yyyy-MM-dd HH:mm:ss")
-        .appendOptional(optional)
+        .append(microsecondsFormatter)
```

Comment:
When using joda date time formatter, a `SSSSSS` pattern will match all `9`, `99`, `999`, etc.
But java cannot. It is strict in nature. for pattern  `SSSSSS`, tt requires it to have 6 `S`s in the date time string, otherwise it will throw an error.
Thats why I am using a `microsecondsFormatter` which supports all possible count of `S`, which is 1-6.

Reply:
ok, but isn't the fact that it's optional means it can be absent (no digit or dot at all)?

Reply:
right. I think I might have made it `append` instead of `appendOptional` by mistake. will correct it

Reply:
fixed